File: impl/src/org/rr0/im/business/actor/BeingImpl.java
package org.rr0.im.business.actor;

import org.rr0.im.business.event.circumstance.Length;
import org.rr0.im.business.event.circumstance.Weight;

import java.util.Locale;

/**
 * Intelligent Being Reference Implementation
 *
 * @author Jerome Beau
 * @version 0.3
 */
public class BeingImpl extends ActorImpl implements Being {
    /**
     * The being's height, in meters
     */
    private Length height;

    /**
     * The being's weight, in kilograms
     */
    private Weight weight;

    public BeingImpl(Identity identity) {
        super(identity);
    }

    public Length getHeight(Locale locale) {
        return height;
    }

    public Weight getWeight() {
        return weight;
    }

}

File: impl/src/org/rr0/im/business/actor/BeingImpl.ts
import {Length} from "../../../../../../../api/src/org/rr0/im/business/event/circumstance/Length";
import {Weight} from "../../../../../../../api/src/org/rr0/im/business/event/circumstance/Weight";
import {Actor} from "./ActorImpl";
import {Identity} from "./IdentityImpl";

/**
 * A single physical and intelligent being.
 */
export class Being extends Actor {
  /**
   * The being's height, in meters
   */
  height: Length;

  /**
   * The being's weight, in kilograms
   */
  weight: Weight;

  constructor(identity: Identity, birth?) {
    super(identity, birth);
  }
}